### PR TITLE
grt: fix usage from underflow

### PR DESCRIPTION
**Pull request by arthurjolo**
Fixes: [#2633](https://github.com/The-OpenROAD-Project/OpenROAD/issues/2633)

At the fastroute after calling the Flute algorthm, the function calls fixEmbeddedTrees(). This method changes the embadded trees, but it doesen't updates the usage for the edges that were removed and the edges that were added. This fix updates this usages after changing the trees.

An exemple can be found on the following test case at the net with ID:15270 edge: 1
[grt-usage-wraps.tar.gz](https://github.com/The-OpenROAD-Project/OpenROAD/files/10278138/grt-usage-wraps.tar.gz)

Before fixing the embadded tree this edge is:

image

after fixing the embadded tree this edge is:

image

But the usage is never added for the new edges, or is decreased for the old edges.

### CODE REVIEW

**Review by eder-matheus — APPROVED** [review]
The update looks fine, and it also fixes a failure in a private design. But let's wait for @maliberty review before merging it.

File: src/grt/src/fastroute/src/maze.cpp
Comment (eder-matheus) [inline src/grt/src/fastroute/src/maze.cpp:182]: Why this extra check here? Same for the code below.

Comment (arthurjolo) [reply src/grt/src/fastroute/src/maze.cpp:182]: I made this based on another code, but i belive it is not necessary.

**Comment by maliberty** [comment]
Why does the loop appear when "fixing"?  It seems like another bug.  It looks like it is just flipping the L segment.

File: src/grt/src/fastroute/src/maze.cpp
Comment (maliberty) [reply src/grt/src/fastroute/src/maze.cpp:182]: what was the other code you based it on?

Comment (arthurjolo) [reply src/grt/src/fastroute/src/maze.cpp:182]: It is based  on the code from the function removeLoops()

**Comment by arthurjolo** [comment]
I belive there is a bug that creates the loop, according to the code it was supose to create the inverse L route, but this loop is generated. I was looking into it today but still didn't find why is this happening.

File: src/grt/src/fastroute/src/maze.cpp
Comment (maliberty) [reply src/grt/src/fastroute/src/maze.cpp:182]: Since loops should never happen in the first place I guess someone was just being extra careful and considering that you might also have zero length edges (which also shouldn't happen).  It isn't harmful so I we can leave it.

Comment (arthurjolo) [reply src/grt/src/fastroute/src/maze.cpp:182]: Ok

**Comment by eder-matheus** [comment]
Secure-ci is green, so I'll merge this PR.

**Pull request merged**
